helper/schema/schema_test.go

```diff
@@ -5556,3 +5556,323 @@ func TestSchemaMapDeepCopy(t *testing.T) {
+	cases := map[string]struct {
+		Key    string
+		Schema map[string]*Schema
+		Config map[string]interface{}
+		Err    bool
+	}{
+
+		"two attributes specified": {
+			Key: "whitelist",
+			Schema: map[string]*Schema{
+				"whitelist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"blacklist"},
+				},
+				"blacklist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"whitelist"},
+				},
+			},
+
+			Config: map[string]interface{}{
+				"whitelist": true,
+				"blacklist": true,
+			},
+			Err: true,
+		},
+
+		"one attributes specified": {
+			Key: "whitelist",
+			Schema: map[string]*Schema{
+				"whitelist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"blacklist"},
+				},
+				"blacklist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"whitelist"},
+				},
+			},
+
+			Config: map[string]interface{}{
+				"whitelist": true,
+			},
+			Err: false,
+		},
+
+		"two attributes of three specified": {
+			Key: "whitelist",
+			Schema: map[string]*Schema{
+				"whitelist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"blacklist", "purplelist"},
+				},
+				"blacklist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"whitelist", "purplelist"},
+				},
+				"purplelist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"whitelist", "blacklist"},
+				},
+			},
+
+			Config: map[string]interface{}{
+				"whitelist":  true,
+				"purplelist": true,
+			},
+			Err: true,
+		},
+
+		"one attributes of three specified": {
+			Key: "whitelist",
+			Schema: map[string]*Schema{
+				"whitelist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"blacklist", "purplelist"},
+				},
+				"blacklist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"whitelist", "purplelist"},
+				},
+				"purplelist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"whitelist", "blacklist"},
+				},
+			},
+
+			Config: map[string]interface{}{
+				"purplelist": true,
+			},
+			Err: false,
+		},
+
+		"no attributes of three specified": {
+			Key: "whitelist",
+			Schema: map[string]*Schema{
+				"whitelist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"blacklist", "purplelist"},
+				},
+				"blacklist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"whitelist", "purplelist"},
+				},
+				"purplelist": &Schema{
+					Type:         TypeBool,
+					Optional:     true,
+					ExactlyOneOf: []string{"whitelist", "blacklist"},
+				},
+			},
+
+			Config: map[string]interface{}{},
+			Err:    true,
+		},
+	}
```

Comment:
Perhaps worth testing some permutations where some or all of the arguments are set to `hcl2shim.UnknownVariableValue`? That sort of thing was particularly troublesome during 0.12 work, partially due to low test coverage illustrating the intended behavior, so I think it would help to test those edges to help make sure we can stay consistent in future changes.

Reply:
`hcl2shim.UnknownVariableValue` has been tested, additional checks were added,. and tests still pass